### ISSUE TITLE
Redistricting informational app

### ISSUE THREAD

**Issue by msarahan** [open-austin/project-ideas#118]
### What problem are we trying to solve?

Politicians cheating people out of their vote with gerrymandering - and the lack of political engagement that allows this to continue.

### Who will benefit (directly and indirectly) from this project?

Hopefully everyone in America.  Directly, we'll be able to inform people quickly how they are being represented, and how redistricting affects that rep's actions.  Indirectly, as redistricting gets more fair, I'm hoping we get some more sane, centered politics from all sides.

### Where can we find any research/data available/articles?

I gave a presentation at Open Austin in October: https://docs.google.com/presentation/d/1w0rpAlB5NQgZDRgT9Tv1g7vPNw2LRwhLkC72xAtkQ6E/edit?usp=sharing

There's a boatload of links in there that are a great place to start.

### What help is needed at this time?

Some initial work was done at the DNC hackathon on Nov 4, 5.  Our concept sketches/pitch are at 
https://docs.google.com/presentation/d/1HXDQ67ZLWtnrXCVowDRLvg6qKtMfgRGzEsBMNmGzO0E/edit?usp=sharing and there's a github org with our work at https://github.com/FairDistricts

### What are the next steps (validation, research, coding, design)?

The UI needs the most work.  I think this means:

* refine design of web interactions
* Some web dev to step people through the process
* More work on the map view to incorporate more data sources, and add the UI interaction, as far as entering in address and having zoom level set

----

#### Checklist for NEW ideas :baby:
Hey, you're official! You're now part of the growing civic hacking community in Austin. Here's a few things to get started (a couple you've probably already done).

- [x] Create this idea issue
- [x] Flesh out the who, where, and what questions above
- [ ] Start the conversation about this new idea on [Slack](https://openaustin-slackin.herokuapp.com/) (this project uses [#general](https://open-austin.slack.com/messages/C036FLME2) )
- [ ] Respond with at least one update on this issue within the next month

#### Checklist for ACTIVE projects :fire:
Let's get this project started! When this idea starts taking off, the Projects Core Team will start helping this project's lead(s) out with project management and connecting you to resources you may need. To get there, please complete and check off the following:

- [x] Post an update at least once a month to this issue. Use [BASEDEF](https://github.com/open-austin/cheatsheet-basedef) for ideas, but it's ok even if your update is just "nothing new happened this month" or "we saw a small increase in traffic to our app this month". If there's no activity for two months, that's no problem, life happens. We'll just label this as `backlog` so others know you'll get back to it when you have the time. If nobody hears from you at all in more than two months, we may mark it as `abandoned` so that others can pick up this idea and run with it.
- [ ] Take 30 minutes to complete [Open Leadership 101](https://mozilla.teachable.com/p/open-leadership-101) 
- [ ] Create a [GitHub repository](https://github.com/new) and Slack channel for work.
- [ ] Create a [README file](https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/write-a-great-project-readme/) in your project repository. This file should help newcomers understand what your project is, why it's important, and kinds of help you're looking for.
- [ ] [Create issues](https://mozilla.github.io/open-leadership-training-series/articles/get-your-project-online/project-set-up-for-collaboration-with-github/#assignment--add-your-first-issue) to describe each task that you plan to do or need help with and how a contributor can get started on that task. You might start and stop a lot, so consider issues as your to-do list.
- [ ] Create a [team](https://help.github.com/articles/creating-a-team/) for your core contributors
  - This will make it easier for you to manage your github repo access. People on a team have the same level of access. Admin access will allow your trusted contributors to make changes as needed. 
  - You can remove and add people to your team as needed. 
  - View some of our teams: [https://github.com/orgs/open-austin/teams](https://github.com/orgs/open-austin/teams)
  - Note: You can also allow collaborators outside of your team and give them more limited access.
- [ ]   Create a [user group](https://get.slack.help/hc/en-us/articles/212906697-User-Groups) in Slack so you can "@" your team all at once
  - You'll need permission from a Slack admin, so just mention @leadership on Slack to get this set up.
  - This will make it easier to get the attention of the core contributors of your project in your project slack channel which may have some spectators or more casual contributors. 
  - People can be added and removed to your group as contributors join and leave. 
- [ ]   Create a [Google Drive](https://drive.google.com), [Dropbox](https://www.dropbox.com/), or other cloud storage to share larger files
  - Github and Data.World are good for code and data, respectively, especially when you need version control. But they're not good for very large files, documentation, articles, etc. A cloud storage option will allow you to easily share, create, and collaborate on documents with your team and help organize ideas and thoughts.
  - Doing this early on can help your team stay organized and to onboard new contributors who wouldn't have access to files you all have shared over email.

#### Checklist for FEATURED Projects :tada:
To have your project FEATURED on [Open-Austin.org](https://open-austin.org/), complete the following documentation. In past projects, well-documented featured projects have more contributions than other projects.

- [ ] In your README, link to the [Open Austin Code of Conduct](https://www.open-austin.org/about/#code-of-conduct) or write your own code of conduct.
- [ ] [Create file: LICENSE](http://choosealicense.com/) to give your project an open license, allowing for sharing and remixing.
- [ ] [Create file: CONTRIBUTING.md](https://github.com/acabunoc/mozsprint-repo-template/blob/master/CONTRIBUTING.md) so others know [how they can contribute](https://mozilla.github.io/open-leadership-training-series/articles/building-communities-of-contributors/write-contributor-guidelines/).
- [ ] Create an easily shareable project management artifact, like a [Civic Tech Canvas](https://cityofaustin.github.io/civic-tech-canvas/) or [Open Canvas](https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/develop-an-open-project-strategy-with-open-canvas/)
- [ ] [Create an issue](https://github.com/open-austin/open-austin.github.io/issues/new) on the open-austin.github.io repo with the title `Add [my project] to projects page`. An Open Austin leader will review this issue and post your project :balloon:
- [ ] [Tell the City of Austin](http://austintexas.gov/page/mobile-app-library). If your idea is in a shareable format and can benefit people around the city, go to that site and follow the instructions on the bottom of the page to showcase your work there.

If you get stuck at any point, feel free to reach out to the leadership team on Slack by adding @leadership to your message. We're here to help you make real changes to our city.

**Comment by joshavenue** [comment]
@msarahan  , I am a UI/UX designer introduced by Vickle O'Dell from Open Austin.
My portfolio : http://zensite.design
My email : joshzensite@gmail.com

**Comment by capndave** [comment]
Do you still need help with this? I've got lots of time at the moment and this sounds worthwhile. Following josh's lead:

My Portfolio: davethompsonisme.com
My Email: capndavet@gmail.com

I'm available to come to the work day tomorrow if you'd like to chat or if there's something for me to work on.

**Comment by msarahan** [comment]
Yes, I'll be at the workday tomorrow, and I'll look forward to chatting.  I don't know exactly what there might be to work on right now, but let's talk about it.  I'm still exploring the technical details of how to plot lots of data together, and trying to get all of the software to do said plotting written.  That plot is going to be cute and useful to nerds, but not a powerful storyteller in and of itself.  What I'd really like help on is the development of websites or similar that use the map plots to tell stories.  These websites will also guide technical requirements of the plot, in terms of what the plot must support for user interactions.

**Comment by capndave** [comment]
Mike - sounds good. I'll be there close to ten but have to run at 11:30.
Hope to see you then

**Comment by tom-montgomery** [comment]
@msarahan I can help with mapping back end if need be

**Comment by msarahan** [comment]
@tom-montgomery sounds great, I'll look forward to talking more with you soon.  I think the next meeting I'll be at is https://www.meetup.com/Open-Austin/events/245784256/ - Feb 26.  Might you also be there?  If not, let's find a different time to talk.

**Comment by msarahan** [comment]
Small project update:

I attended a gerrymandering workshop last weekend:
 https://www.ma.utexas.edu/users/blumberg/gerrymandering.html

I was part of the hackathon.  There's a lot of facets to the problem.  One common uniting theme among many of the district analysis and simulation is representation of the subunits (census blocks, VTD's, precincts, etc.) in a adjacency graph (i.e. nodes and edges).  This is what lets the mathematicians and computer scientists play their games to say whether districts are fair, and compute alternate districts based on some optimization parameters.  It also lets them call politicians' bluff on the reasons why they say they drew maps in a given way.  The simulations can be run with those stated reasons, and if the given map is highly unlikely to be generated with those stated reasons, but likely to be generated with other reasons in place, then the politicians are in trouble.

Anyway, I hacked on tools to help go from the polygon representation of those units to the adjacency graph.  The repo is at https://github.com/gerrymandr/adjacency-graphs

I also connected with James Santucci, a developer at Azavea, who is reviving DistrictBuilder.  It is possible that some of the work that has been done so far with mapping here may feed into DistrictBuilder.

**Comment by jplaut** [comment]
Hi @msarahan,

I'm a total geo data nerd and recently just put together a map of the new 2018 Pennsylvania districts that I hope to use to educate PA residents of changes that might effect them. I love the idea of creating a tool that people can use to see redistricting in action and possibly learn about the reasoning behind it.

In terms of background, I'm a fullstack we developer with a strong focus on Ruby on Rails and ReactJS. I have lots of experience with data processing, mostly using Node and Python. I also have a good amount of experience with open mapping libraries and Mapbox in order to visualize data.

You can take a look at a district map I developed here: https://vote.foundation/find-my-district

I'll be at the hack day tomorrow and will be looking for projects to work on. Hope to see you there, but if not, maybe we can connect outside of the group. Let me know what you think.

**Comment by msarahan** [comment]
@jplaut I'm not sure if I'll be around tomorrow.  My parents are in town visiting.  There is a group of people interested in this sort of problem meeting roughly weekly right now.  Our next meeting is probably going to be on Thursday.  Email me at msarahan@gmail.com and I'll copy you on any communication there.

**Comment by jplaut** [comment]
Hey all,

TL;DR I created a tool for PA voters to visualize the change to their district, given the new 2018 redistricting, and find their new reps based on that. Check it out and let me know what you think!

https://vote.foundation/find-my-district-pa

For those interested:

For the data, I pulled the most recent shapefiles from the PA gov't website (http://www.redistricting.state.pa.us/), transformed it, and loaded it into a geospatial Postgres DB capable of geocoding addresses and reverse geocoding lat/lng to the new PA districts. To my knowledge, it's the first interactive tool of its kind to do this.

I have an API layer to the DB that I would be glad to provide access to if anyone is interested in using the data. I also built the platform to be to be flexible enough to handle future redistricting shapefiles. I hope to be able to use this to visualize, and thus inform, voters as to how redistricting will effect them personally.

The API is in Ruby on Rails, DB is Postgres with the Postgis extension, the map was created using Mapbox, which is built on Leaflet.

Let me know what you think!

**Comment by mscarey** [comment]
I believe this project was more about visualizing the arguments about gerrymandering that were current prior to Justice Kennedy's retirement, and the playing field has apparently shifted a little since then. There was a really good update about it on the [fivethirtyeight podcast](https://fivethirtyeight.com/features/politics-podcast-can-statistics-solve-partisan-gerrymandering/). I'm marking this project inactive.
